python/lsst/meas/base/plugins.py

```diff
@@ -137,23 +137,39 @@ def __init__(self, config, name, schema, metadata):
         SingleFramePlugin.__init__(self, config, name, schema, metadata)
         self.keyProbability = schema.addField(name + "_value", type="D",
                                               doc="Set to 1 for extended sources, 0 for point sources.")
+        self.keyFlag = schema.addField(name + "_flag", type="Flag",
+                                              doc="Set to 1 for any fatal failure.")
 
     def measure(self, measRecord, exposure):
         modelFlux = measRecord.getModelFlux()
         modelFluxErr = measRecord.getModelFluxErr()
+        modelFluxFlag = measRecord.getModelFluxFlag()
         psfFlux = measRecord.getPsfFlux()
         psfFluxErr = measRecord.getPsfFluxErr()
-        flux1 = self.config.fluxRatio*modelFlux + self.config.modelErrFactor*modelFluxErr
-        flux2 = psfFlux + self.config.psfErrFactor*psfFluxErr
-        if flux1 < flux2:
-            measRecord.set(self.keyProbability, 0.0)
+        psfFluxFlag = measRecord.getPsfFluxFlag()
+        flux1 = self.config.fluxRatio*modelFlux
+        if not self.config.modelErrFactor == 0:
+            flux1 += self.config.modelErrFactor*modelFluxErr
+        flux2 = psfFlux
+        if not self.config.psfErrFactor == 0:
+            flux2 += self.config.psfErrFactor*psfFluxErr
```

Comment:
The logic here appears to have got more complicated for no very clear reason. If `self.config.modelErrFactor == 0`, then adding something multiplied by it has no effect (and the same for `self.config.psfErrFactor`, of course). What are the `if` statements adding?